### PR TITLE
Add unit test for CancelAllDialogs and fix eventName and eventValue null

### PR DESCRIPTION
**Pull request by tomlm**
#3534 
Fixes CancelAllDialogs.eventName and eventValue is valid to be null

### CODE REVIEW

File: libraries/Microsoft.Bot.Builder.Dialogs/Recognizer.cs [hunk 1]
```diff
@@ -47,7 +47,7 @@ public class Recognizer
         /// <param name="telemetryProperties">Additional properties to be logged to telemetry with the LuisResult event.</param>
         /// <param name="telemetryMetrics">Additional metrics to be logged to telemetry with the LuisResult event.</param>
         /// <returns>Analysis of utterance.</returns>
-        public virtual async Task<RecognizerResult> RecognizeAsync(DialogContext dialogContext, Activity activity, CancellationToken cancellationToken = default, Dictionary<string, string> telemetryProperties = null, Dictionary<string, double> telemetryMetrics = null)
+        public virtual Task<RecognizerResult> RecognizeAsync(DialogContext dialogContext, Activity activity, CancellationToken cancellationToken = default, Dictionary<string, string> telemetryProperties = null, Dictionary<string, double> telemetryMetrics = null)
```
Comment (cleemullins) [inline libraries/Microsoft.Bot.Builder.Dialogs/Recognizer.cs]: I don't understand this, and it seems like an odd one. Can you elaborate?
